### PR TITLE
Make function not access outer-state 

### PR DESCRIPTION
**Pull request by live-wire**
- Minor refactor to use functionality upstream

### CODE REVIEW

File: pkg/flink/handler.go [hunk 1]
```diff
@@ -158,10 +158,16 @@ func (h flinkResourceHandler) OnAbort(ctx context.Context, tCtx pluginsCore.Task
...
 }
 
-func FlinkClusterTaskLogs(ctx context.Context, flinkClusterName string, flinkClusterNamespace string) ([]*core.TaskLog, error) {
+type FlinkTaskLogs struct {
+	ClusterName string
+	Namespace   string
+	Config      *Config
+}
+
+func FlinkClusterTaskLogs(ctx context.Context, flinkTaskLogs FlinkTaskLogs) ([]*core.TaskLog, error) {
```
Comment (regadas) [inline pkg/flink/handler.go:167]: Let's have `*Config` as an argument, and could `FlinkTaskLogsInput` be a better name? similar to `tasklog.Input`?

**Review by regadas — CHANGES_REQUESTED** [review]

**Comment by live-wire** [comment]
@regadas made the suggested changes 👍 
Please have a look

File: pkg/flink/handler.go [hunk 2]
```diff
@@ -158,10 +158,14 @@ func (h flinkResourceHandler) OnAbort(ctx context.Context, tCtx pluginsCore.Task
 	return abortBehavior, nil
 }
 
-func FlinkClusterTaskLogs(ctx context.Context, flinkClusterName string, flinkClusterNamespace string) ([]*core.TaskLog, error) {
+type FlinkTaskLogsInput struct {
+	ClusterName string
+	Namespace   string
+}
+
+func FlinkClusterTaskLogs(ctx context.Context, fi FlinkTaskLogsInput, config *Config) ([]*core.TaskLog, error) {
```
Comment (regadas) [inline pkg/flink/handler.go:166]:
```suggestion
func FlinkClusterTaskLogs(ctx context.Context, config *Config, fi FlinkTaskLogsInput) ([]*core.TaskLog, error) {
```

File: pkg/flink/handler.go [hunk 3]
```diff
@@ -187,7 +191,10 @@ func FlinkClusterTaskLogs(ctx context.Context, flinkClusterName string, flinkClu
 func flinkClusterTaskInfo(ctx context.Context, flinkCluster *flinkOp.FlinkCluster) (*pluginsCore.TaskInfo, error) {
 	var taskLogs []*core.TaskLog
 
-	tl, err := FlinkClusterTaskLogs(ctx, flinkCluster.Name, flinkCluster.Namespace)
+	tl, err := FlinkClusterTaskLogs(ctx, FlinkTaskLogsInput{
+		ClusterName: flinkCluster.Name,
+		Namespace:   flinkCluster.Namespace,
+	}, GetFlinkConfig())
```
Comment (regadas) [inline pkg/flink/handler.go:197]:
```suggestion
	tl, err := FlinkClusterTaskLogs(ctx, GetFlinkConfig(), FlinkTaskLogsInput{
		ClusterName: flinkCluster.Name,
		Namespace:   flinkCluster.Namespace,
	})
```

**Review by regadas — APPROVED** [review]
🚀

File: pkg/flink/handler.go (hunk 3 above)
Comment (regadas) [reply pkg/flink/handler.go]: :art: I did a minor param reorder here.

**Pull request merged**